clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -147,6 +152,10 @@ public class PrototypeAsyncConsumer<K, V> implements Consumer<K, V> {
     // to keep from repeatedly scanning subscriptions in poll(), cache the result during metadata updates
     private boolean cachedSubscriptionHasAllFetchPositions;
     private final WakeupTrigger wakeupTrigger = new WakeupTrigger();
+    private final AtomicBoolean fencedInstance = new AtomicBoolean(false);
```

Comment:
```suggestion
    private volatile boolean isFenced = false;
```

Any reason this needs to be anything other than a `boolean`?